codespan-reporting::Files not implemented for codespan::Files<String>

When attempting to upgrade Gleam to 0.11.0 of codespan and codespan-reporting, I got the following error. 

```
error[E0277]: the trait bound `codespan::Files<std::string::String>: codespan_reporting::files::Files<'_>` is not satisfied
  --> src/diagnostic.rs:59:5
   |
59 |     emit(&mut buffer, &config, &files, &diagnostic).unwrap();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `codespan_reporting::files::Files<'_>` is not implemented for `codespan::Files<std::string::String>`
```

I guess I'm either missing something or this was overlooked.

--- a/codespan/CHANGELOG.md
+++ b/codespan/CHANGELOG.md
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+
+-   Removed an erroneous feature gate from the implementation of
+    `codespan_reporting::Files` for `codespan::Files`.
+
 ## [0.11.0] - 2020-11-30
 
 There is now a [code of conduct](https://github.com/brendanzab/codespan/blob/master/CODE_OF_CONDUCT.md)

--- a/codespan/Cargo.toml
+++ b/codespan/Cargo.toml
@@ -14,5 +14,8 @@ edition = "2018"
 codespan-reporting = { path = "../codespan-reporting", version = "0.11.0" }
 serde = { version = "1", optional = true, features = ["derive"]}
 
+[dev-dependencies]
+termcolor = "1"
+
 [features]
 serialization = ["serde", "codespan-reporting/serialization"]

--- a/codespan/src/file.rs
+++ b/codespan/src/file.rs
@@ -424,4 +424,27 @@ mod test {
 
         assert_eq!(line_sources, ["foo\n", "bar\r\n", "\n", "baz"],);
     }
+
+    #[test]
+    fn interoperability() {
+        extern crate termcolor;
+        use codespan_reporting::{diagnostic::*, term::emit};
+        use termcolor::{ColorChoice, StandardStream};
+
+        let mut files = Files::<String>::new();
+        let file_id = files.add("test", TEST_SOURCE.to_owned());
+
+        let diagnostic = Diagnostic::note()
+            .with_message("middle")
+            .with_labels(vec![Label::primary(file_id, 4..7).with_message("middle")]);
+
+        let config = codespan_reporting::term::Config::default();
+        emit(
+            &mut StandardStream::stdout(ColorChoice::Auto),
+            &config,
+            &files,
+            &diagnostic,
+        )
+        .unwrap();
+    }
 }

--- a/codespan/src/file.rs
+++ b/codespan/src/file.rs
@@ -221,7 +221,6 @@ where
     }
 }
 
-#[cfg(feature = "reporting")]
 impl<'a, Source> codespan_reporting::files::Files<'a> for Files<Source>
 where
     Source: AsRef<str>,

